### ISSUE TITLE
Relocate top bar and open scenes

### ISSUE THREAD

**Issue by karai17** [godotengine/godot-proposals#6923]
### Describe the project you are working on

New to Godot, exploring a simple tutorial while I familiarize myself with the engine.

### Describe the problem or limitation you are having in your project

The way that different UI panels interact and control each other is not very intuitive. A bit of time familiarizing myself with the UI has helped understand most of the interactions, but some remain confusing.

![image](https://github.com/godotengine/godot-proposals/assets/590332/fa2e23ce-20c5-4dec-bc97-59a5a3d0cbd6)

Specifically, the top bar (2D, 3D, Script, AssetLib buttons) controls what is shown in the main panel, but it is physically disconnected from the panel. Opening scenes creates new tabs that are connected to the main panel, but they do not control the main panel in any way. This creates a lot of confusion when trying to navigate the UI.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I think it would be better for the top bar to be physically connected to the main panel as 4 tabs rather than its own floating UI element.

![image](https://github.com/godotengine/godot-proposals/assets/590332/5bdcac01-ada7-404e-9815-16777cc118cd)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Replace the scene tabs with context tabs for the 2D, 3D, Script, and AssetLib views that display in the main panel so that the views and their respective tabs are physically connected.

The downside to this change would be that there isn't really a good place to list all of the open scenes. The scene panel is the obvious choice, but it's a thin side panel so a horizontal list of tabs wouldn't become cumbersome to use with only a couple scenes open.

An option for the scene tabs could be to add an open scenes panel where the highlighted scene displays its node tree in the scenes menu.

![image](https://github.com/godotengine/godot-proposals/assets/590332/af2a43af-cbad-411b-9ad6-e8e1f52acb2d)

This would be a bit similar to VSCode's "Open Editors" panel that displays a verticle list of open files.

![image](https://github.com/godotengine/godot-proposals/assets/590332/440c05da-050c-48a7-bfd1-db3197ecf85a)

With this proposed change, control and interaction between panels cascades upward (and then to the right). Filesystem -> Open Scenes -> Scene -> Inspector

![image](https://github.com/godotengine/godot-proposals/assets/590332/5db497d9-4753-4c93-baa8-21d91670e4dc)

Personally I think that cascading downwards is more intuitive, but I also understand that the Scene panel has a much higher priority over Filesystem or the proposed Opened Scenes panels, so cascading upward is probably the best option.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Ideally, any UI change here wouldn't add any extra clicks or submenus. This change may require the default sizes of the left panels to change to accomidate an additional panel, but that new panel could be relatively short (by default) to minimize any extra scrolling needed in the other panels.

### Is there a reason why this should be core and not an add-on in the asset library?

core UI change

**Comment by Zireael07** [comment]
Hmmm. I can't say I got opinion either way, but it's an interesting thing to ponder

**Comment by karai17** [comment]
Another option would be to blend the open scenes and scene panels such that the active scene displays its node tree and the inactive scenes are collapsed.

**Comment by yosoyfreeman** [comment]
I think the Godot way of handling this is really smart and intuitive once you understand how it works.

> Specifically, the top bar (2D, 3D, Script, AssetLib buttons) controls what is shown in the main panel, but it is physically disconnected from the panel.

It have to be disconnected, otherwise you could not keep a work mode (2d, scripting...) and switch between scenes at the same time without adding another panel just for opened scenes.

> Opening scenes creates new tabs that are connected to the main panel, but they do not control the main panel in any way.

They control what is shown in the main panel.

I created this simple example:

![imagen](https://github.com/godotengine/godot-proposals/assets/64601424/5c1ffb94-4d93-4da7-a982-f40eb31b07e2)

The scene here control the panel in various ways:

- If you click a 2d node it will automatically change to 2d view. If you select the 3D node, it does the same with 3D view. 
- If you change between a 3D a 2D scene it will do this too.
- Even if you just select the parent , which can be whatever a 3D or a 2D node and switch between 2D and 3D view it will show you the corresponding elements of the scene.


For scripting, even tho the main panel do not change, the important things that you need to interact with the world still change.  you still can switch between scenes to access node paths, see hierarchies or make references in code via drag and drop.
![imagen](https://github.com/godotengine/godot-proposals/assets/64601424/ff50137f-25cd-4e9c-9afd-aa1c44157c88)
![imagen](https://github.com/godotengine/godot-proposals/assets/64601424/8707247c-e123-449a-b911-00ea298f0166)

(This apply to the inspector properties too which you will need to read per scene to access them via code)

**Comment by karai17** [comment]
> They control what is shown in the main panel.
That is not true. Those tabs have no connection whatsoever to what is displayed on themain panel. Here is a screenshot of my key scene being open but my main panel has a world script open. If I change which scene tab I am on, it does change the main panel, but you can then click something else and be out of sync again. That's bad UX because it is inconsistant.

![image](https://github.com/godotengine/godot-proposals/assets/590332/5639d654-c559-4e30-8d94-89bbf697884d)

**Comment by zinnschlag** [comment]
> That is not true.

Not quite. It is 3/4 true. The tabs control the main panel when in 3D, 2D or Control mode. Only the script editor is not affected. Which actually makes sense, since scripts aren't scenes and often are also not directly linked with scenes (particularly not in a 1-1 relation).

I do agree that the current layout of the UI kinda obscures this fact and beginners seem to get confused by that. Experienced that myself and we have also seen several proposals/bug reports from users confused about the UI.
